scripts/tasks.sh

```diff
@@ -570,18 +570,31 @@ function migrate_pvcs() {
     fi
     CEPH_DISK_USAGE_TOTAL=$(kubectl exec -n rook-ceph deployment/$ROOK_CEPH_EXEC_TARGET -- ceph df | grep TOTAL | awk '{ print $8$9 }')
 
-    # check that longhorn is healthy
-    LONGHORN_NODES_STATUS=$(kubectl get nodes.longhorn.io -n longhorn-system -o=jsonpath='{.items[*].status.conditions.Ready.status}')
-    LONGHORN_NODES_SCHEDULABLE=$(kubectl get nodes.longhorn.io -n longhorn-system -o=jsonpath='{.items[*].status.conditions.Schedulable.status}')
-    pat="^True( True)*$" # match "True", "True True" etc but not "False True" or ""
-    if [[ $LONGHORN_NODES_STATUS =~ $pat ]] && [[ $LONGHORN_NODES_SCHEDULABLE =~ $pat ]]; then
-        echo "All Longhorn nodes are ready and schedulable"
-    else
-        if [ "$SKIP_LONGHORN_HEALTH_CHECKS" = "1" ]; then
-            echo "Continuing with unhealthy Longhorn due to skip-longhorn-health-checks flag"
+
+    local non_ceph_storage_class_detected
+    if kubectl get namespace longhorn-system &>/dev/null; then
+        non_ceph_storage_class_detected="longhorn"
+        # check that longhorn is healthy
```

Comment:
can we move the code below here in the if statement to its own function `longhorn_provisioner_is_healthy`